### ISSUE TITLE
panic upon upgrading to canary on Linux

### ISSUE THREAD

**Issue by 00ff0000red** [denoland/deno#9478]
stdout & stack trace:
```
> RUST_BACKTRACE=full ./.deno/bin/deno upgrade --canary;
Looking up latest version
Found latest version d6c05b09dd7cbaba0fcae65929a2c2dd55e9d994
Checking https://dl.deno.land/canary/d6c05b09dd7cbaba0fcae65929a2c2dd55e9d994/deno-x86_64-unknown-linux-gnu.zip
Download has been found
Deno is upgrading to version d6c05b09dd7cbaba0fcae65929a2c2dd55e9d994
Archive:  /tmp/.tmphGitPu/deno..zip
  inflating: deno                    
/tmp/.tmphGitPu/deno: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.tmphGitPu/deno)
thread 'main' panicked at 'assertion failed: output.status.success()', cli/tools/upgrade.rs:255:3
stack backtrace:
   0:     0x55d567c0cb83 - std::backtrace_rs::backtrace::libunwind::trace::h04d12fdcddff82aa
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/../../backtrace/src/backtrace/libunwind.rs:100:5
   1:     0x55d567c0cb83 - std::backtrace_rs::backtrace::trace_unsynchronized::h1459b974b6fbe5e1
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55d567c0cb83 - std::sys_common::backtrace::_print_fmt::h9b8396a669123d95
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55d567c0cb83 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::he009dcaaa75eed60
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x55d56773f00c - core::fmt::write::h77b4746b0dea1dd3
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/fmt/mod.rs:1078:17
   5:     0x55d567c0c421 - std::io::Write::write_fmt::heb7e50902e98831c
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/io/mod.rs:1518:15
   6:     0x55d567c0bde5 - std::sys_common::backtrace::_print::h2d880c9e69a21be9
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x55d567c0bde5 - std::sys_common::backtrace::print::h5f02b1bb49f36879
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x55d567c0bde5 - std::panicking::default_hook::{{closure}}::h658e288a7a809b29
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:208:50
   9:     0x55d567c0b3c5 - std::panicking::default_hook::hb52d73f0da9a4bb8
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:227:9
  10:     0x55d567c0b3c5 - std::panicking::rust_panic_with_hook::hfe7e1c684e3e6462
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:593:17
  11:     0x55d5679776ec - std::panicking::begin_panic::{{closure}}::hcb184a9ed52d5de5
  12:     0x55d5679776c6 - std::sys_common::backtrace::__rust_end_short_backtrace::haf2505beafd41248
  13:     0x55d56797777d - std::panicking::begin_panic::h3ea500739663b54d
  14:     0x55d5676cefdf - deno::tools::upgrade::upgrade_command::{{closure}}::h5b31c30d7dff30ef
  15:     0x55d56760dd29 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h1a35a640e19b0e86
  16:     0x55d5678ae446 - tokio::runtime::Runtime::block_on::h9bb2ea13d88c1ea4
  17:     0x55d5676e6d67 - deno::main::he860fdd0651ddee0
  18:     0x55d5674d36aa - std::sys_common::backtrace::__rust_begin_short_backtrace::h12856b605070de86
  19:     0x55d5676f3dd6 - main
  20:     0x7fc98d405bf7 - __libc_start_main
  21:     0x55d56746dc7a - _start
  22:                0x0 - <unknown>
```

Seems like my system is missing `GLIBC_2.29`?

**Comment by bnoordhuis** [comment]
Yeah. I think the binaries are built on a Ubuntu 20.04 system with glibc 2.31. You're using an older distro (like Ubuntu 18.04 with glibc 2.27) and that's why it can't find the symbols.

Glibc symbol version issues were a perennial topic for Node.js that we solved by building on the oldest distro we could reasonably support, but that was a never-ending battle to actually keep it building.

It's possible with some assembly-level hacking to pin glibc symbols to an older version. It seems there are only three right now, all from libm, so perhaps that's a good short-term solution:
```
$ nm deno | grep '@@GLIBC_2[.]2[0-9]' | sort
                 U exp@@GLIBC_2.29
                 U log2@@GLIBC_2.29
                 U log@@GLIBC_2.29
```

**Comment by 00ff0000red** [comment]
Ah, I'll just try instaling glibc 2.31 on my system.

**Comment by bnoordhuis** [comment]
This has been fixed in v1.7.4, by building the release on an older system.
